pkg/servicequota/servicequota.go

```diff
@@ -0,0 +1,89 @@
+// Package servicequota check if you have enough resources in aws before cluster creation starts
+package servicequota
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/service/servicequotas"
+	"github.com/hjson/hjson-go"
+	"github.com/oslokommune/okctl/pkg/api/okctl.io/v1alpha1"
+	"github.com/tidwall/gjson"
+)
+
+// Usage defines what we need to know about a service quota
+type Usage interface {
+	Count() (int, error)
+	Quota() (int, error)
+	Required() int
+	CheckAvailability() error
+	Description() string
+}
+
+// CheckQuotas check if we have enough resources for given cloud provider
+func CheckQuotas(provider v1alpha1.CloudProvider) error {
```

Comment:
Here we should accept a list of the `Usage` interfaces, rather than the `v1alpha1.CloudProvider`, e.g.:

```go
func CheckQuotas(checks []Usage) error { ... }
```

We can then iterate over the checks to determine if they are good or not.

Reply:
Done :+1: